command/command.go

```diff
@@ -346,6 +346,37 @@ func Prepare(conf *config.Config) (*mackerel.API, *mackerel.Host, error) {
 	return api, host, nil
 }
 
+// RunOnce collects specs and metrics, then output them to stdout.
+func RunOnce(conf *config.Config) {
+	hostname, meta, interfaces, err := collectHostSpecs()
+	if err != nil {
+		logger.Errorf("While collecting host specs: %s", err)
+		return
+	}
+	ag := &agent.Agent{
+		MetricsGenerators: metricsGenerators(conf),
+		PluginGenerators:  pluginGenerators(conf),
+	}
+	graphdefs := ag.CollectGraphDefsOfPlugins()
+	logger.Infof("Collecting metrics may take one minutes.")
+	metrics := ag.CollectMetrics(time.Now())
+	payload := map[string]interface{}{
+		"host": map[string]interface{}{
+			"name":          hostname,
+			"meta":          meta,
+			"interfaces":    interfaces,
+			"roleFullnames": conf.Roles,
+		},
+		"metrics": metrics,
+	}
```

Comment:
This data structure reflects what `mackerel.UpdateHost()` sents to the API server, so this part should be moved to another function which would also be used by UpdateHost, not duplicating code.